sdk/eventhub/azure-eventhub/tests/eventhub_preparer.py

```diff
@@ -0,0 +1,309 @@
+# The resource preparer.
+
+import functools
+
+from azure.mgmt.eventhub import EventHubManagementClient
+from azure.mgmt.eventhub.models import Eventhub, AccessRights
+
+from azure_devtools.scenario_tests.exceptions import AzureTestError
+
+from devtools_testutils import (
+    ResourceGroupPreparer, AzureMgmtPreparer, FakeResource
+)
+
+from devtools_testutils.resource_testcase import RESOURCE_GROUP_PARAM
+
+EVENTHUB_DEFAULT_AUTH_RULE_NAME = 'RootManageSharedAccessKey'
+EVENTHUB_NAMESPACE_PARAM = 'eventhub_namespace'
+EVENTHUB_PARAM = 'eventhub'
+EVENTHUB_AUTHORIZATION_RULE_PARAM = 'eventhub_authorization_rule'
+
+# Service Bus Namespace Preparer and its shorthand decorator
+class EventHubNamespacePreparer(AzureMgmtPreparer):
+    def __init__(self,
+                 name_prefix='',
+                 use_cache=False,
+                 sku='Standard', location='westus',
+                 parameter_name=EVENTHUB_NAMESPACE_PARAM,
+                 resource_group_parameter_name=RESOURCE_GROUP_PARAM,
+                 disable_recording=True, playback_fake_resource=None,
+                 client_kwargs=None, random_name_enabled=True):
+        super(EventHubNamespacePreparer, self).__init__(name_prefix, 24,
+                                                          random_name_enabled=random_name_enabled,
+                                                          disable_recording=disable_recording,
+                                                          playback_fake_resource=playback_fake_resource,
+                                                          client_kwargs=client_kwargs)
+        self.location = location
+        self.sku = sku
+        self.resource_group_parameter_name = resource_group_parameter_name
+        self.parameter_name = parameter_name
+        self.connection_string = ''
+        if random_name_enabled:
+            self.resource_moniker = self.name_prefix + "ehname"
+
+        self.set_cache(use_cache, sku, location)
+
+    def create_resource(self, name, **kwargs):
+        if self.is_live:
+            self.client = self.create_mgmt_client(EventHubManagementClient)  # type: EventHubManagementClient
+            group = self._get_resource_group(**kwargs)
+            namespace_async_operation = self.client.namespaces.begin_create_or_update(
+                group.name,
+                name,
+                {
+                    'sku': {'name': self.sku},
+                    'location': self.location,
+                }
+            )
+            self.resource = namespace_async_operation.result()
+
+            key = self.client.namespaces.list_keys(group.name, name, EVENTHUB_DEFAULT_AUTH_RULE_NAME)
+            self.connection_string = key.primary_connection_string
+            self.key_name = key.key_name
+            self.primary_key = key.primary_key
+
+            self.test_class_instance.scrubber.register_name_pair(
```

Comment:
Don't need scrubber any more :)  Should be handled by the unified scrubber component in the core test harness, so long as you're only creating one resource in the preparer and assuming you're using the name that's provided. (and in this case you're doing both)

Reply:
Removed the scrubber